### PR TITLE
Initialize data version control for managing test images

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR implements the solution for storing baseline test images proposed in #5724. Only a file describing the contents of a directory with baseline images are stored in the git repository; the actual PS file will be stored in the DAGsHub repository (e.g., https://dagshub.com/GenericMappingTools/gmt/src/manage-tests-dvc-refactor/test/baseline/api). While by default DAGsHub will store all versions of the images that are pushed to the remote storage, these can be easily garbage collected from either the user's local cache and the remote storage (if we reach the 10 GB limit) without breaking the git history. 

These are the specific changes in this PR:

- Initialize dvc in the repository
- Add dvc as a dependency for developers
- Add dvc to the CI test workflow
- Migrate test/api and test/blockmean .PS files to dvc
- Add instructions for migrating baseline images to dvc to the contributing guide

Addresses #5724 

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

File: doc/rst/source/devdocs/contributing.rst
Comment (weiji14) [inline doc/rst/source/devdocs/contributing.rst:194]:
```suggestion
#. If you will need to push baseline images to the remote, ask a GMT maintainer to add you as a collaborator on
   `DAGsHub <https://dagshub.com/GenericMappingTools/gmt>`_.
```

File: doc/rst/source/devdocs/contributing.rst
Comment (weiji14) [inline doc/rst/source/devdocs/contributing.rst:219]:
```suggestion
#. Remove the test image from git tracking (``git rm --cached <module>/<test-image>.ps``).
```

File: .github/workflows/tests.yml
Comment (weiji14) [inline .github/workflows/tests.yml:124]:
```suggestion
        run: dvc pull
```

This can be on one line, unless you want to have `ls -lhR test/api test/baseline ...` lines too.

**Review by weiji14 — COMMENTED** [review]
Sorry for the long delay! I only have a few minor comments for now, the dvc setup and instructions seem mostly ok for me, but I'll need to run a proper `git clone` of this branch and do some proper tests in the coming days. Disk space on my 10 year old laptop isn't great for big repos, so I'll need to use some university resources :slightly_smiling_face:

**Comment by meghanrjones** [comment]
> Sorry for the long delay! I only have a few minor comments for now, the dvc setup and instructions seem mostly ok for me, but I'll need to run a proper `git clone` of this branch and do some proper tests in the coming days. Disk space on my 10 year old laptop isn't great for big repos, so I'll need to use some university resources 🙂

No worries on the timing - I really appreciate your willingness to give this a review!

**Comment by meghanrjones** [comment]
Would anyone be willing to give this a review shortly? I think it would be great to make a decision about whether to go with dvc before we update the baseline images for the ~25 failing tests.

Here are some generally instructions for testing out the changes:
1. Install dvc (https://dvc.org/doc/install)
2. Checkout branch: `git checkout manage-tests-dvc-refactor`
3. Pull baseline images stored in dvc: `dvc pull`
4. Run the tests in your regular way (e.g., https://docs.generic-mapping-tools.org/dev/devdocs/contributing.html#running-tests).

Afterwards, you would need to do some cleanup (this wouldn't be necessary when switching branches if this PR gets merged in):
1. Checkout the other branch (e.g., `git checkout master`)
2. Navigate to the base of the repository (e.g., `gtop` if using aliases)
3. Delete dvc related files (`rm -rf .dvc` and `rm -rf test/baseline`)

One note: migrating all the tests files to dvc would initially add ~100 MB to the repository size for people who run GMT tests locally (see https://github.com/GenericMappingTools/gmt/issues/5724 for details) if we don't also delete git history (which I would prefer avoiding). For anyone who doesn't run `dvc pull`, this change would have a negligible impact on repository size. To avoid this step increase in size for those who run the tests, I think we can only migrate baseline images when tests fail due to code updates or when new tests are added.

**Comment by PaulWessel** [comment]
I will give it a go following the instructions - will slack if running into troubles!

**Comment by meghanrjones** [comment]
Thanks Paul!

**Comment by PaulWessel** [comment]
Was able to successfully do all 7 steps.

**Review by weiji14 — APPROVED** [review]
Haven't found the time to run the full tests, but the documented `dvc` steps look ok to me, so won't hold this back any longer. :rocket:

**Comment by meghanrjones** [comment]
@PaulWessel do you approve of merging this in now? If so, I could use dvc for the postscript files in https://github.com/GenericMappingTools/gmt/pull/6199 and I could update the reference images for the failing tests using dvc.

**Comment by PaulWessel** [comment]
Yes, I think that would be great.  Go ahead.

**Pull request merged**
